### ISSUE TITLE
table-req-header does not correctly interpret <thead

### ISSUE THREAD

**Issue by hejmartin** [htmllint/htmllint#197]
The following markup is not recognized as a table with a header:

```html
<table>
  <caption>Caption</caption>
  <thead>
    <tr>
      <th>Header</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Data</td>
    </tr>
  </tbody>
</table>
```

It yields the error message `"table must have a header for accessibility"`

I'm using htmllint 0.6.0 via grunt-htmllint 0.2.7

**Comment by YodasWs** [comment]
Yes, the problem can be seen even when the `<thead>` is on the next line:

```html
<table>
<thead>
⋮
```

but you add an empty `<thead>` to the same line as the `<table>` opening tag and then the rule will pass:

```html
<table><thead></thead>
  <caption>Caption</caption>
  <thead>
    <tr>
  ⋮
```

**Comment by YodasWs** [comment]
Although gulp-html-lint uses htmllint 0.3.0, I want to bump this issue.

**Comment by jrabbe** [comment]
I have a similar issue with the following snippet. It seems that any valid element inside `<table>` before <thead>` or the `<tr>` with the `<th>` elements breaks.

```html
<table>
  <colgroup>
    <col>
    <col>
  </colgroup>
  <thead>
  <tr>
    <th>Column 1</th>
    <th>Column 2</th>
  </tr>
  </thead>
  <tbody>
  <tr>
    <td>Value 1</td>
    <td>Value 2</td>
  </tr>
  </tbody>
</table>
```

**Comment by YodasWs** [comment]
@jrabbe,
Yes, I discovered working to fix this with #200, the problem is actually that you are including text (whitespace in this case) between the different tags between `<table>` and `<thead>`&hellip;

**Comment by mlochbaum** [comment]
Fixed by #200.
